include/sparrow/arrow_interface/arrow_array_schema_utils.hpp

```diff
@@ -0,0 +1,264 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <algorithm>
+#include <optional>
+#include <ranges>
+#include <tuple>
+#include <type_traits>
+#include <vector>
+
+#include "sparrow/memory.hpp"
+#include "sparrow/mp_utils.hpp"
+
+namespace sparrow
+{
+    /**
+     * Get the size of a range, a tuple or an optional.
+     * If the range is a sized range, the size is obtained by calling size().
+     * If the range is a tuple, the size is obtained by calling tuple_size_v.
+     * If the optional has a value, the size is obtained by calling ssize() on the value.
+     *
+     * @tparam T The type of the value.
+     * @param value The value.
+     * @return The size of the value.
+     */
+    template <class T>
+    constexpr int64_t ssize(const T& value);
+
+    /**
+     * Get a raw pointer from a smart pointer, a range, an object or a pointer.
+     *
+     * @tparam T The type of the pointer to obtain.
+     * @tparam U The type of the variable.
+     * @param var The variable.
+     * @return A raw pointer.
+     *          If the variable is a smart pointer, the pointer is obtained by calling get().
+     *          If the variable is a range, the pointer is obtained by calling data().
+     *          If the variable is a pointer, the pointer is returned as is.
+     *          If the variable is an object, the pointer is returned by calling the address-of operator.
+     */
+    template <typename T, typename U>
+    T* get_raw_ptr(U& var);
+
+    /**
+     * Create a vector of pointers to elements from a range.
+     * The range must be a non-view range.
```

Comment:
```suggestion
     * Requirement: The provided range must own it's elements.
```

By stating it's a requirement we imply it's UB to do otherwise.

Reply:
Ok done